src/libs/browser-storage.js

```diff
@@ -31,12 +28,74 @@ export const setDynamic = (storage, key, value) => {
   }
 }
 
-export const removeDynamic = (storage, key) => {
+export const getStatic = (storage, key) => {
+  return maybeParseJSON(storage.getItem(key))
+}
+
+export const setStatic = (storage, key, value) => {
+  if (value === undefined) {
+    storage.removeItem(key)
+  } else {
+    forceSetItem(storage, key, JSON.stringify(value))
+  }
+}
+
+export const listenStatic = (storage, key, fn) => {
+  window.addEventListener('storage', e => {
+    if (e.storageArea === storage && e.key === key) {
+      fn(maybeParseJSON(e.newValue))
+    }
+  })
+}
+
+export const getDynamic = (storage, key) => {
+  const storageKey = `dynamic-storage/${key}`
+  const data = maybeParseJSON(storage.getItem(storageKey))
+  return data && data.value
+}
+
+export const setDynamic = (storage, key, value) => {
+  const storageKey = `dynamic-storage/${key}`
+  if (value === undefined) {
+    storage.removeItem(storageKey)
+  } else {
+    forceSetItem(storage, storageKey, JSON.stringify({ timestamp: Date.now(), value }))
+  }
+}
+
+export const listenDynamic = (storage, key, fn) => {
   const storageKey = `dynamic-storage/${key}`
-  storage.removeItem(storageKey)
+  window.addEventListener('storage', e => {
+    if (e.storageArea === storage && e.key === storageKey) {
+      const data = maybeParseJSON(e.newValue)
+      fn(data && data.value)
+    }
+  })
 }
 
-const withUserPrefix = key => `${getUser().id}/${key}`
+/**
+ * Returns a stateful object that manages the given storage location.
+ * Implements an atom-compaible interface, and can be passed to useAtom.
+ */
+
+export const staticStorageSlot = (storage, key) => {
```

Comment:
Should we allow the key to have a user prefix? This could be done with an added option of withUserPrefix.

Reply:
With the current state of the code, that would require some additional restructuring. The user id can change over time, so the listener would need to take that into account.

One way to do that would be to move all this logic into a hook, which could use e.g. `useAtom` to monitor the user id. That would work fine, but it would mean this layer could only be used from within a component, potentially making it harder to reuse in other ways.

The other approach would be to use RxJS and define the whole thing as a 'stream' (which could be derived from the 'stream' of user ids). We're trending in that direction, but not quite there yet.

Reply:
What about exposing the withUserPrefix function? I am thinking about the case where we need to identify specific user has timed out - would there be a way to do this?

Reply:
I think the plan is to have a single entry which is an object that maps user ids to timeouts. That way components can connect to this one slot, as well as the `authStore`, and perform the lookup to find the timeout for the current user.